### PR TITLE
Attach manifest to enable UTF-8 on Windows

### LINKED ISSUE #683: Windows cjxl.exe cannot handle Japenes/Chinese/Russian unicode filenames
I have a log of pictures with russian, japanese and chinese filenames.
Examples: 3d-art-красивые-картинки-art-пираты-6497178.jpeg or фото-Природа-озеро-горы-6946340.jpeg
When I rename them using the CMD shell:
ren фото-Природа-озеро-горы-6946340.jpeg 00000temp.jpeg
I can convert them and rename it back afterwards. See attached screenshot.
It does not happen für typical WESTERN unicode filenames with German Umlaut like ÖÄÜ.

JPEG-XL-cjxl exe-unicode-issue

That exact source picture on the internet:
http://joyreactor.cc/post/4943227

Cannot test that for *nix, don't have a linux at home any more (last was about 2015). But since the normale cmd copy / xcopy / robocopy and all other windows command line tools can handle it it should be possible for the windows version of CJXL too.

kind regards,

Joachim Otahal

### PR DESCRIPTION
**Pull request by AlifianK**
### Description
Attach manifest to enable reading files with Unicode characters name (eg. Japanese, Russian, etc). Note that this only works for Windows 10 version 1903 and later. See [this link](https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page) for more information. This is taken from @scurest and @kmilos contribution, and uses [libavif](https://github.com/AOMediaCodec/libavif) for its inspiration. Tested on MSVC with cl, clang-cl, clang-gnu frontend, and MSYS2 with ucrt64 (GCC toolchain) and clang64 (LLVM toolchain).
Fixes #683.

### Pull Request Checklist

- [x] **CLA Signed**: Have you signed the [Contributor License Agreement](https://code.google.com/legal/individual-cla-v1.0.html) (individual or corporate, as appropriate)? Only contributions from signed contributors can be accepted.
- [x] **Authors**: Have you considered adding your name to the [AUTHORS](AUTHORS) file?
- [x] **Code Style**: Have you ensured your code adheres to the project's coding style guidelines? You can use `./ci.sh lint` for automatic code formatting.

### CODE REVIEW

File: tools/CMakeLists.txt [hunk 1]
```diff
@@ -336,6 +336,18 @@ foreach(BINARY IN LISTS TOOL_BINARIES)
...
+  # Attach manifest that tells Windows to use UTF-8 for eg. fopen
+  if(WIN32)
+    # Accommodate cl, clang-cl and clang with GNU-like command
+    if(NOT MINGW)
+	  target_sources(${BINARY} PRIVATE utf8.manifest)
+	# Since CMake in MINGW doesn't support linking
+	# .manifest file, do it with .rc file
+	elseif(MINGW)
+	  target_sources(${BINARY} PRIVATE utf8.rc)
+	endif()
```
Comment (scurest) [inline tools/CMakeLists.txt:349]: Indentation appears to be messed up here.

Comment (AlifianK) [reply tools/CMakeLists.txt:349]: Yeah, let me see if I can fix it.

Comment (scurest) [reply tools/CMakeLists.txt]: It still seems to be wrong. I think your editor is turning spaces into tabs.

Comment (AlifianK) [reply tools/CMakeLists.txt]: Should be fixed now, had to tweak my VSCode settings.

**Review by mo271 — APPROVED** [review]
LGTM, thanks!

**Review by sboukortt — APPROVED** [review]

**Comment by Theo1996** [comment]
Would be nice ig it didnt drop support for anything before Malware 10 version 1903

**Comment by sboukortt** [comment]
> Would be nice ig it didnt drop support for anything before Malware 10 version 1903
 
1. No need for the inflammatory language.
 
2. It shouldn’t: https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#activeCodePage

   > This element was first added in Windows 10 version 1903 (May 2019 Update). You can declare this property and target/run on earlier Windows builds

**Comment by AlifianK** [comment]
> Let me rephrase, maybe add something that works before 1903? Because I run it and still cant read image data , in my case it has some chinese and Tiếng Việt chars)

There's no 'something' if you use Windows that is older than Windows 10 version 1903, other than patching the code to use -W API, but that would mean a complete overhaul of the entire codebase.
You could, however, use a PowerShell script to temporarily rename the name of the files.
```powershell
Get-ChildItem *.png,*.jpg | ForEach-Object {$name = $_.BaseName;Rename-Item -Path $_ -NewName ($_ -replace $_.BaseName,'a');cjxl -e 9 -d 1 "a$($_.Extension)" "a.jxl";Rename-Item -Path "a$($_.Extension)" -NewName "$name$($_.Extension)";Rename-Item -Path 'a.jxl' -NewName "$name.jxl"}
```

**Comment by K4sum1** [comment]
> > it didnt drop support
> 
> It doesn't _drop_ anything, as there was no support for Unicode (i.e. legacy -W API conversion) for Windows in libjxl tools to begin with. This at least _adds_ UTF-8 for Windows 10 1903 and later. On earlier Windows versions it continues working just the same as today, i.e. ASCII only.

He's not completely wrong, but like it's only an issue once you get to XP. The official release doesn't work on pre-7, but a build with clang can run just fine on XP, but it requires a change to the manifest.

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
    <application>
      <windowsSettings>
        <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
      </windowsSettings>
    </application>
  </compatibility>
</assembly>
```

**Comment by K4sum1** [comment]
Nevermind, apparently this breaks UTF-8 support on 10.
